crates/storage/src/vm_storage.rs

```diff
@@ -344,11 +344,13 @@ where
         let mut state_key = Bytes32::zeroed();
 
         let mut results = Vec::new();
-        for _ in 0..range {
+        for i in 0..range {
```

Comment:
couldn't we alter the range from 1..range instead of checking every time if the i !=0 ?

Reply:
The problem is that for i == 0 we still want to do the operations below in the loop. We could try to start at `start_key - 1` but that will cause issue in case start key is the lowest possible. I never see good solution in these cases :/ If you have an idea you can try to apply it